packages/babel-traverse/src/scope/lib/renamer.js

```diff
@@ -97,6 +116,30 @@ export default class Renamer {
     );
   }
 
+  parameterDefaultHasOuterBinding(path, state) {
+    const { oldName, scope } = state;
+    const params = path.get("params");
+    for (let i = 0; i < params.length; i++) {
+      const param = params[i];
+      if (param.isAssignmentPattern()) {
```

Comment:
This is not enough; someone could also do this: (which is currently broken anyway)

```js
let name;

function fn({ [name]: value }) {
  let name;
}
```

or

```js
let name;

function fn({ key = name }) {
  let name;
}
```